components/common/pkg/prometheus/metrics.go

```diff
@@ -72,17 +86,56 @@ func (m *metricsServer) initializeServer() {
 }
 
 // Initialize The MetricsServer's Logger
-func (m *metricsServer) initializeLogger() {
+func (m *MetricsServer) initializeLogger() {
 
 	// Create A New Zap Production Logger
 	logger, err := zap.NewProduction()
 	if err != nil {
-		log.Fatalf("Failed To Create Zap Production Logger: %v", err)
+		log.Logger().Fatal("Failed To Create Zap Production Logger: %v", zap.Error(err))
 	}
 	if logger == nil {
-		log.Fatalf("Created Nil Zap Production Logger")
+		log.Logger().Fatal("Created Nil Zap Production Logger")
 	}
 
 	// Set The Initialized Logger
 	m.logger = logger
 }
+
+// Observe librdkafka Metrics
+func (m *MetricsServer) Observe(stats string) {
+
```

Comment:
I'm missing how/where this is called - if we're implementing some prometheus interface can we add an explicit check of that to make it clear?  Maybe a comment such as.... "Called by..." as a breadcrumb ?

Reply:
called by the other half of PR in channel/dispatcher  ; )